### PR TITLE
Feature/map coin listeners

### PR DESCRIPTION
**Pull request by gauthierboeshertz**
The coins are now updated with the database, when a coin is removed in the database it will also be removed from the map, and when a coin is removed from the map it will remove it from the database if it has been collected.

### CODE REVIEW

**Review by Alvaro2112 — APPROVED** [review]

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java [hunk 1]
```diff
@@ -112,23 +112,75 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                }else{
+                }
+            } else {
+                Log.e(Game.class.getSimpleName(), task.getException().getMessage());
+            }
+        });
+    }
 
+    /**
+     * Adds a database proxy to the map
```
Comment (Warhedgehog) [inline app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:142]: I think the comment and method name are poorly phrased. If I'm not mistaken, it seems that this method get the game from the db.

**Review by Warhedgehog — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java [hunk 2]
```diff
@@ -53,6 +59,28 @@ public static void setPersistence(){
...
+    public Game getGame() {
+        String name = "Game";
+        Player host = new Player(3, "Bob", 0);
+        int maxPlayerCount = 2;
+        List<Riddle> riddles = new ArrayList<>();
+        riddles.add(new Riddle("yes?", "blue", "green", "yellow", "brown", "a"));
+        List<Coin> coins = new ArrayList<>();
+        coins.add(new Coin(0., 0., 1));
+        Location location = new Location("LocationManager#GPS_PROVIDER");
+        location.setLatitude(37.4219473);
```
[the remark this answers is not in the document]
Comment (gauthierboeshertz) [reply app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java:78]: I am not sure what it is I just copy pasted from the GameInstrumented test ;)

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java [hunk 3]
```diff
@@ -112,23 +112,75 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                game = proxyG.getGameFromTaskSnapshot(task);
 
-            if(player.equals(game.getHost())){
-                game.setCoins(localPlayer.getLocallyAvailableCoins(), false);
-            }else{
+                if(player.equals(game.getHost())){
+                    game.setCoins(localPlayer.getLocallyAvailableCoins(), false);
+                    addCoinsListener();
+                }else{
+                }
```
[the remark this answers is not in the document]
Comment (Tesa27) [reply app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:134]: That will be filled up in my PR.

File: app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tesa27) [reply app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java:78]: Oh, okay this is in test once again my bad

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gauthierboeshertz) [reply app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:146]: I deleted it

**Review by Warhedgehog — APPROVED** [review]

**Pull request merged**